### PR TITLE
Fix activate scripts for dpcpp compiler

### PR DESCRIPTION
**Pull request by isuruf**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by napetrov** [comment]
Hi @isuruf - is there content for this PR? Not sure I understand everything here

**Comment by oleksandr-pavlyk** [comment]
Yes, a context would be nice. 

Instead of specifying `--gcc-toolchain`, `--sysroot`, and `--target` in $CFLAGS, we [leverage](https://github.com/IntelPython/dpctl/blob/master/conda-recipe/build.sh#L6-L9) support for compiler configuration files:

```bash
# Intel LLVM must cooperate with compiler and sysroot from conda
echo "--gcc-toolchain=${BUILD_PREFIX} --sysroot=${BUILD_PREFIX}/${HOST}/sysroot -target ${HOST}" > icpx_for_conda.cfg
export ICPXCFG="$(pwd)/icpx_for_conda.cfg"
export ICXCFG="$(pwd)/icpx_for_conda.cfg"
```

Perhaps this could be used in the compiler activation script.

**Comment by isuruf** [comment]
This is for https://github.com/conda-forge/onednn-feedstock/pull/53. Yeah we can use that script.

File: recipe/scripts/activate-dpcpp.sh [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+fi
+
+export CC=icx
+export CXX=icpx
+export ICPXCFG="$BUILD_PREFIX/bin/$HOST-icpx.cfg"
+export ICXCFG="$BUILD_PREFIX/bin/$HOST-icx.cfg"
+
+if [ $? -ne 0 ]; then
+  echo "ERROR: $(_get_sourced_filename) failed, see above for details"
+else
```
Comment (oleksandr-pavlyk) [inline recipe/scripts/activate-dpcpp.sh:36]: How does the {{_get_sourced_filename}} get called? Forgive my scripting ignorance

File: recipe/repack.sh [hunk 2]
```diff
@@ -21,4 +21,7 @@ fi
 if [[ "$PKG_NAME" == "dpcpp_impl_linux-64" ]]; then
   # Not sure why these libraries are there, but seems to be unused.
   rm -f $PREFIX/lib/libffi.so*
+  HOST=x86_64-conda-linux-gnu
+  echo "--gcc-toolchain=${PREFIX} --sysroot=${PREFIX}/${HOST}/sysroot -target ${HOST}" > ${PREFIX}/bin/$HOST-icpx.cfg
+  echo "--gcc-toolchain=${PREFIX} --sysroot=${PREFIX}/${HOST}/sysroot -target ${HOST}" > ${PREFIX}/bin/$HOST-icx.cfg
```
Comment (oleksandr-pavlyk) [inline recipe/repack.sh:26]: I thin it should be `${BUILD_PREFIX}/bin/$HOST-icpx.cfg`.

Comment (isuruf) [reply recipe/repack.sh:26]: This is when the compiler itself is being built. PREFIX at compiler build time becomes BUILD_PREFIX at downstream package build time.

**Review by oleksandr-pavlyk — APPROVED** [review]

**Comment by isuruf** [comment]
@napetrov can you review and merge?

**Review by napetrov — APPROVED** [review]

**Pull request merged**